components/server/src/workspace/workspace-service.ts

```diff
@@ -119,8 +122,55 @@ export class WorkspaceService {
         return workspace;
     }
 
-    async getWorkspace(userId: string, workspaceId: string): Promise<Workspace> {
-        return this.doGetWorkspace(userId, workspaceId);
+    async getWorkspace(
+        userId: string,
+        workspaceId: string,
+        check: (workspace: Workspace, instance?: WorkspaceInstance) => Promise<void> = async () => {},
```

Comment:
seems like we don't need the check function, as the result can be checked on the call site

Reply:
Nice, this turns out to be superfluous indeed. But also I deem them super easy to remove post rollout.

Reply:
These last argument check functions were not meant to be a general pattern but really only an exceptional hack we use when the old guards need to be colled somewhere within the logic.
I'd really love to not introduce anything in our services layer that is not strictly necessary.

Reply:
> These last argument check functions were not meant to be a general pattern but really only an exceptional hack

Totally agree here. But the way we went about to do this transition (internal, without interface breakage), I think it's easier to